server/graphql/v2/object/Host.js

```diff
@@ -1137,6 +1145,130 @@ export const GraphQLHost = new GraphQLObjectType({
           };
         },
       },
+      hostedAccounts: {
+        type: new GraphQLNonNull(GraphQLAccountCollection),
+        description: 'Returns a list of accounts hosted by this host',
+        args: {
+          ...getCollectionArgs({ limit: 100, offset: 0 }),
+          accountType: { type: new GraphQLList(GraphQLAccountType) },
+          isApproved: {
+            type: GraphQLBoolean,
+            description: 'Filter on (un)approved collectives',
+            defaultValue: true,
+          },
+          isFrozen: {
+            type: GraphQLBoolean,
+            description: 'Filter on frozen accounts',
+            defaultValue: false,
+          },
+          isUnhosted: {
+            type: GraphQLBoolean,
+            description: 'Filter on unhosted accounts',
+            defaultValue: false,
+          },
+          hostFeesStructure: {
+            type: GraphQLHostFeeStructure,
+            description: 'Filters on the Host fees structure applied to this account',
+          },
+          searchTerm: {
+            type: GraphQLString,
+            description:
+              'A term to search membership. Searches in collective tags, name, slug, members description and role.',
+          },
+          orderBy: {
+            type: GraphQLOrderByInput,
+            description: 'Order of the results',
+          },
+        },
+        async resolve(host, args, req) {
+          const isAdmin = req.remoteUser?.isAdmin(host.id);
+          if (!isAdmin) {
+            throw new Unauthorized('You need to be logged in as an admin of the host to see its hosted accounts');
+          }
+
+          const where = {
+            HostCollectiveId: host.id,
+          };
+
+          if (args.accountType && args.accountType.length > 0) {
+            where.type = {
+              [Op.in]: args.accountType.map(value => AccountTypeToModelMapping[value]),
+            };
+          }
+
+          if (!isNil(args.isFrozen)) {
+            if (args.isFrozen) {
+              set(where, `data.features.${FEATURE.ALL}`, false);
+            } else {
+              set(where, `data.features.${FEATURE.ALL}`, { [Op.is]: null });
+            }
+          }
+
+          if (args.hostFeesStructure) {
+            if (args.hostFeesStructure === HOST_FEE_STRUCTURE.DEFAULT) {
+              where.data = { useCustomHostFee: { [Op.not]: true } };
+            } else if (args.hostFeesStructure === HOST_FEE_STRUCTURE.CUSTOM_FEE) {
+              where.data = { useCustomHostFee: true };
+            } else if (args.hostFeesStructure === HOST_FEE_STRUCTURE.MONTHLY_RETAINER) {
+              throw new ValidationFailed('The MONTHLY_RETAINER fees structure is not supported yet');
+            }
+          }
+
+          if (args.isUnhosted) {
+            const unhostedActivities = await models.Activity.findAll({
+              where: { type: 'collective.unhosted', HostCollectiveId: host.id },
+              attributes: ['CollectiveId'],
+            });
```

Comment:
Not a blocker, and not sure how true it is, but I'm wondering if using the Members table to find deleted memeberships as in the previous solution might be faster than searching in the larger Activities table

Reply:
Hey @gustavlrsn, we have two problems with the Members table I'm trying to avoid:
1. A host Member record is created before the host application is approved, which causes this to list collectives that applied and were rejected.
2. You can have multiple instances of the same membership deleted, this caused some collectives to appear multiple times.